docs/tutorial/quickstart.md

```diff
@@ -0,0 +1,355 @@
+
+At the command line, use [`go get`](https://golang.org/cmd/go/#hdr-Add_dependencies_to_current_module_and_install_them) to add the `oras.land/oras-go/v2` module as a dependency for your module. Use a dot argument to mean "get dependencies for code in the current directory."
+
+```
+go get .
+go: added github.com/opencontainers/go-digest v1.0.0
+go: added github.com/opencontainers/image-spec v1.1.0
+go: added golang.org/x/sync v0.6.0
+go: added oras.land/oras-go/v2 v2.5.0
+```
+
+Run the code.
+```
+go run .
+```
+
+You should see the tags in the repository.
+
+## Push a layer to the repository
+
+All referenced layers must exist in the repository before a manifest can be pushed. The following code snippet demonstrates how to push a manifest layer using the [(*Repository) Push](https://pkg.go.dev/oras.land/oras-go/v2@v2.5.0/registry/remote#Repository.Push) method. Paste the code into `main.go` after the last section.
+
+```
+// 3. push a layer to the repository
+layer := []byte("example manifest layer")
+layerDescriptor := content.NewDescriptorFromBytes(v1.MediaTypeImageLayer, layer)
+err = repo.Push(ctx, layerDescriptor, bytes.NewReader(layer))
+if err != nil {
+	panic(err)
+}
+fmt.Println("Pushed manifest layer:", layerDescriptor.Digest)
+```
+
+## Push a manifest to the repository with the tag "quickstart"
+
+The following code snippet demonstrates how to pack a manifest and push it to the repository with the tag "quickstart" using the [PackManifest](https://pkg.go.dev/oras.land/oras-go/v2@v2.5.0#PackManifest) and the [(*Repository) Tag](https://pkg.go.dev/oras.land/oras-go/v2@v2.5.0/registry/remote#Repository.Tag) methods. Paste the code into `main.go` after the last section.
+
+```
+// 4. Push a manifest to the repository with the tag "quickstart"
+packOpts := oras.PackManifestOptions{
+	Layers: []v1.Descriptor{layerDescriptor},
+}
+artifactType := "application/vnd.example+type"
+desc, err := oras.PackManifest(ctx, repo, oras.PackManifestVersion1_1, artifactType, packOpts)
+if err != nil {
+	panic(err)
+}
+tag := "quickstart"
+err = repo.Tag(ctx, desc, tag)
+if err != nil {
+	panic(err)
+}
+fmt.Println("Pushed and tagged manifest")
+```
+
+## Fetch the manifest from the repository by tag
+
+The following code snippet demonstrates how to fetch a manifest from the repository by its tag. First, the descriptor associated with the tag is resolved using [(*Repository) Resolve](https://pkg.go.dev/oras.land/oras-go/v2@v2.5.0/registry/remote#Repository.Resolve). Then, the content is fetched and read using [FetchAll](https://pkg.go.dev/oras.land/oras-go/v2@v2.5.0/content#FetchAll) from the [content](https://pkg.go.dev/oras.land/oras-go/v2@v2.5.0/content) package. Paste the code into `main.go` after the last section.
+
+```
+// 5. Fetch the manifest from the repository by tag
+desc, err = repo.Resolve(ctx, tag)
+if err != nil {
+	panic(err)
+}
+fetchedManifestContent, err := content.FetchAll(ctx, repo, desc)
+if err != nil {
+	panic(err)
+}
```

Comment:
This is technically correct for all targets. However, it is not the best practice for the best performance.

Since we are given the context of remote repository, consider [FetchReference](https://pkg.go.dev/oras.land/oras-go/v2/registry/remote#Repository.FetchReference).

Reply:
Consider using [FetchBytes](https://pkg.go.dev/oras.land/oras-go/v2#FetchBytes) here instead

Reply:
Updated with [FetchBytes](https://pkg.go.dev/oras.land/oras-go/v2#FetchBytes) here